### DISCUSSION TITLE
RE: cannot unmarshal !!map into []*targetgroup.Group"

### DISCUSSION THREAD

**Discussion by venkatakrishna4444** [prometheus/prometheus#14827] [Q&A]
we are trying to scrape matric from snmp.yml, the yml placed in /etc/prometeus/snmp/snmp_netapp.yml

-----------------------
  - job_name: 'netapp'
    scrape_interval: 120s
    scrape_timeout: 120s
    metrics_path: /snmp
    params:
      module: [IF-MIB]
    static_configs:
      - targets:
          - 10.16.18.17
          - 10.16.18.42
        labels:
          region: 'oregon'
    relabel_configs:
      - source_labels: [__address__]
        target_label: __param_target
      - source_labels: [__param_target]
        target_label: instance
      - target_label: __address__
        replacement: 127.0.0.1:9116  # URL as shown on the UI

    file_sd_configs:
      - files:
          - /etc/prometheus/snmp/snmp_netapp.yml  # Path to your SNMPv3 configuration file

-----------------------------------

but we are facing error 

Sep 04 12:32:09 saasprometh01 prometheus[20441]: ts=2024-09-04T12:32:09.196Z caller=file.go:337 level=error component="discovery m  anager scrape" discovery=file msg="Error reading file" path=/etc/prometheus/snmp/snmp_netapp.yml err="yaml: unmarshal errors:\n  line 2: cannot   unmarshal !!map into []*targetgroup.Group"

**Comment by beorn7** [comment]
`/etc/prometheus/snmp/snmp_netapp.yml` doesn't seem to have the expected format. Note that `file_sd_configs` means that you list targets to scrape in the referenced file. It is not the file you use to configure the target itself.
